### ISSUE TITLE
LeaveOneOutGroup cross-validation do not use test split

### ISSUE THREAD

**Issue by jpcaico** [pycaret/pycaret#867]
Hello, I'm trying to use leaveoneoutgroup cross validation from scikitlearn.

I need to train my model on X-y samples and test in y samples

In my specific examples I have 32 distinct groups
So it should train on 31 and test on 1, and so on..

```
from sklearn.model_selection import LeaveOneGroupOut
cv = LeaveOneGroupOut()

regression_setup = setup(data = data,
                    target = 'target', session_id=123, 
                    numeric_imputation = 'median',
                    categorical_imputation = 'constant',
                    normalize = True,
                    fold_strategy = cv,
                    fold_groups = 'column_name_w_group'
                        )
```
In the setup, it shows that the data is being splitted in train and test set (I dont want), and that 10 folds are generated. It is increasing my model performance because it is putting same group on test and train.

![image](https://user-images.githubusercontent.com/27935809/99827052-eaf43280-2b37-11eb-8867-4d9634afc542.png)

But at the same time, it shows while 'compare_model' is running 32 folds (expected) and not the 10 in the setup info.

![image](https://user-images.githubusercontent.com/27935809/99827224-23940c00-2b38-11eb-9a78-cafdf9e282ea.png)

Question: How should I configure setup to let my model apply leave one out grouped cv properly (train in all groups -1 and test in the out group) without creating train and test and mixing the groups?

**Comment by pycaret** [comment]
@Yard1 Can you take a look at this one?

**Comment by Yard1** [comment]
Train-test splitting is always going to happen as that's needed for PyCaret to work. You can set the size of the train set with `train_size` parameter - you could set it to, eg. `0.99`. Alternatively, you can pass a dummy train dataframe with `train_data` param, but you need to make sure it has all of the same columns as test data. The number of folds in setup is just a graphical oversight - as you noticed, it works correctly when the models are created.

**Comment by atwahsz** [comment]
@Yard1 can you create an example of how to setup LeaveOneGroupOut
where the training set never sees the folded group . , in pycaret when passing 0.99 it seems to test on the leftover 1%

**Comment by Yard1** [comment]
I'm not sure I understand what you mean. Can you explain it in more detail @atwahsz ?

**Comment by atwahsz** [comment]
![image](https://user-images.githubusercontent.com/24771158/106911583-5dc28200-6713-11eb-9fcd-570625f9f302.png)


![image](https://user-images.githubusercontent.com/24771158/106911371-1dfb9a80-6713-11eb-9486-6d7bc18d11db.png)

![image](https://user-images.githubusercontent.com/24771158/106911405-2b188980-6713-11eb-8df7-b7e8364fcde2.png)


![image](https://user-images.githubusercontent.com/24771158/106911456-38ce0f00-6713-11eb-9ffb-ed74ad552f6b.png)


see the results here . 
does that represent a blind test in a full group ?? no splitting was done on the group ?

the same results are not achieved when splitting manually and training and testing on a seprate group
in this image the following data is used
![image](https://user-images.githubusercontent.com/24771158/106912329-f8bb5c00-6713-11eb-886f-609544ba4e0e.png)

![image](https://user-images.githubusercontent.com/24771158/106912158-d0cbf880-6713-11eb-99ab-aad145996b3d.png)

**Comment by atwahsz** [comment]
how do i make sure when im using leaveonegrupout that in each fold , whole group was taken out for testing , and the training was done on the rest of the groups ?

**Comment by Yard1** [comment]
It looks to be working correctly, though? There are 10 groups, so the CV generator creates 10 folds. Everything looks to be working as expected.

**Comment by atwahsz** [comment]
the question is what is it testing on ? what is the size of the test , is it a whole group ? or 1% of a group ?

**Comment by Yard1** [comment]
This is cross validation, so the 1% test set is not used here. From reading the documentation of LeaveOneGroupOut, it uses one group as the test set and the remaining groups as the train set and does so for each group.

**Comment by atwahsz** [comment]
The thing is when done manually on all groups . Same results are not achieved
